src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -882,7 +882,9 @@ internal IndexCreationOptions GetIndexCreationOptions(object indexDefinition, In
             if (indexDef != null)
             {
                 differences = existingIndex.Definition.Compare(indexDef);
-                //Index definition.state can be different from index state so we need to check both
+                //We need to check differences between old and new definition
```

Comment:
The key is that we need to figure out the difference between the _in-memory_ state and the definition state.
And not only compare the old def with the new def.